### PR TITLE
Feature ETP-81: Add dynamic menu

### PR DESCRIPTION
**Pull request by LuuchoRocha**
Feature ETP-81: Add dynamic menu

### CODE REVIEW

File: packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx [hunk 1]
```diff
@@ -1,94 +1,82 @@
...
-}) => {
-  // Constants
-  const isMainSection = level === 0;
-  const isSelected =
-    section.isSelected ||
-    section.subSections?.some((subsection: Section) => subsection.isSelected);
-
-  // States
+const DrawerSection = ({ item, onClick }: DrawerSectionProps<Menu>) => {
+  const isMainSection = item.type === 'folder';
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx:10]: string in constant

File: packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx [hunk 2]
```diff
@@ -1,94 +1,82 @@
...
+  const isSelected = false;
   const [expanded, setExpanded] = useState(false);
 
-  // Handlers
-  const handleClick = () => {
-    if (section.subSections) {
+  const handleClick = useCallback(() => {
+    if (item.type === 'folder') {
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx:15]: same

File: packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx [hunk 3]
```diff
@@ -1,94 +1,82 @@
...
+    () => ({
+      ...styles.drawerSectionBox,
+      backgroundColor:
+        isMainSection && isSelected
+          ? theme.palette.baselineColor.neutral[10]
+          : 'transparent',
+      padding:
+        isMainSection && !!item.submenu && isSelected ? '0.5rem' : undefined,
+    }),
+    [isMainSection, isSelected, item.submenu],
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx:32]: Is it necessary to useMemo to change styles?

Is the calculation required too heavy?

File: packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx [hunk 4]
```diff
@@ -1,94 +1,82 @@
...
-        {open &&
-          section.subSections &&
-          (expanded ? <ExpandLess /> : <ExpandMore />)}
+  const submenuStyle = useMemo(
+    () => ({
+      ...styles.contentBox,
+      alignItems: 'flex-start',
+    }),
+    [],
+  );
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx:61]: same

File: packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx [hunk 5]
```diff
@@ -1,94 +1,82 @@
...
+    [isSelected],
+  );
+
+  const titleStyle = useMemo(
+    () => ({
+      ...styles.listItemInnerContentText,
+      justifyContent: 'flex-start',
+    }),
+    [],
+  );
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx:53]: Using useMemo just to add a style does not seem to me a correct way to use it. Also by passing an empty dependency array, the value will only be calculated once when the component is mounted, which is the same as doing this:

const titleStyle = {
...styles.listItemInnerContentText,
      justifyContent: 'flex-start',
};

File: packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx [hunk 6]
```diff
@@ -1,94 +1,82 @@
...
+    () => ({
+      ...styles.listItemButton,
+      ...(isSelected ? styles.listItemButtonSelected : undefined),
+      ...styles.listItemContentText,
+      borderRadius: '0.5rem',
+      fontSize: '1.5rem',
+      justifyContent: 'space-between',
+    }),
+    [isSelected],
+  );
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx:45]: same

File: packages/ComponentLibrary/src/components/Drawer/DrawerSubsection/index.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+  const handleClick = useCallback(() => {
+    if (item.type === 'folder') {
+      setExpanded(prev => !prev);
+    } else if (item.type === 'window') {
+      onClick(`/window/${item.windowId}`);
+    }
+  }, [item.type, item.windowId, onClick]);
+
+  return (
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Drawer/DrawerSubsection/index.tsx:23]: having many applications of conditional styles is a bit difficult to read and maintain.

I would extrude styles to variables or functions, for example: 

const drawerSectionStyle = {
  ...styles.drawerSectionBox,
  backgroundColor: isMainSection && isSelected
    ? theme.palette.baselineColor.neutral[10]
    : 'transparent',
  padding: isMainSection && !!item.submenu && isSelected ? '0.5rem' : undefined,
};

File: packages/ComponentLibrary/src/components/Drawer/Header/index.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  logo,
+  open,
+  onClick,
+}: {
+  title: string;
+  logo: string;
+  open?: boolean;
+  onClick: () => unknown;
+}) {
+  const sx = useMemo(
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Drawer/Header/index.tsx:19]: the same as using useMemo to change styles.

File: packages/ComponentLibrary/src/components/Drawer/index.tsx [hunk 9]
```diff
@@ -1,164 +1,49 @@
...
-      setExpandedSections({});
-    }
-  }, [open]);
+import { DrawerProps } from './types';
+import DrawerHeader from './Header';
+
+const Drawer = ({ items, logo, title, onClick }: DrawerProps) => {
+  const [open, setOpen] = useState<boolean>(true);
+  const handleHeaderClick = useCallback(() => setOpen(prev => !prev), []);
+  const paperProps = useMemo(
```
Comment (leandroallemandismf) [inline packages/ComponentLibrary/src/components/Drawer/index.tsx:11]: same

File: packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx (hunk 2 above)
Comment (sjcuello) [inline packages/ComponentLibrary/src/components/Drawer/DrawerSection/index.tsx:15]: Use the enum instead the string.
This is repeated in the whole PR

File: packages/EtendoHookBinder/src/contexts/metadata.tsx [hunk 10]
```diff
@@ -1,14 +1,25 @@
...
 
 const ctx = { getWindow: Metadata.getWindow, getColumns: Metadata.getColumns };
 
 export const MetadataContext = createContext(ctx);
 
 export default function MetadataProvider(props: React.PropsWithChildren) {
+  const [ready, setReady] = useState(false);
+
+  useEffect(() => {
+    const f = async () => {
```
Comment (sjcuello) [inline packages/EtendoHookBinder/src/contexts/metadata.tsx:12]: use a mnemotechnic  name for naming functions and constants

File: packages/MainUI/src/index.css [hunk 11]
```diff
@@ -6,3 +6,21 @@
...
+
+#root {
+  height: 100%;
+  width: 100%;
+  background: linear-gradient(180deg, #C6D3FF 30%, #F5F6FA 100%);
+}
+
+.animated-width {
+  transition: width 300ms cubic-bezier(0.3, 0.1, 0.2, 1) !important;
+}
```
Comment (leandroallemandismf) [inline packages/MainUI/src/index.css:26]: add a line break at the end

Comment (sjcuello) [inline packages/MainUI/src/index.css:26]: missing return at final of this file

**Review by leandroallemandismf — APPROVED** [review]

File: packages/EtendoHookBinder/src/api/metadata.ts [hunk 12]
```diff
@@ -6,6 +6,9 @@ import * as Etendo from './types';
 
 export type { Etendo };
 
+// function sleep(delay: number) {
```
Comment (sjcuello) [inline packages/EtendoHookBinder/src/api/metadata.ts:9]: Don't leave comments like this

**Review by sjcuello — DISMISSED** [review]
Small comments to fix!
Well done

**Review by sjcuello — APPROVED** [review]
image

**Pull request merged**